dali/kernels/imgproc/resample/separable_cpu.h

```diff
@@ -113,35 +113,48 @@ struct SeparableResampleCPU  {
            const Input &input,
            const ResamplingParams2D &params) {
     auto &desc = setup.desc;
+
+    Surface2D<const InputElement> in_ROI = as_surface_HWC(input);
```

Comment:
I would suggest moving this ROI setup to some function of surface.
So something along the lines of:
```suggestion
    auto in_ROI = as_surface_HWC(input).get_roi(width, height, start_coords.x);
```